spacy/tokens/doc.pyx

```diff
@@ -1462,6 +1462,146 @@ cdef class Doc:
+        for a, annot in enumerate(annotations):
+            if annot is not None:
+                if len(annot) != len(words):
+                    raise ValueError(Errors.E189)
+                headings.append(possible_headings[a])
+
+        # If there are any other annotations, set them.
+        if headings:
+            attrs = self.to_array(headings)
+            j = 0
+
+            for annot in annotations:
+                if annot:
+                    if annot is not token_annotations[HEAD]["values"]:
+                        for i in range(len(annot)):
+                            if annot[i] is not None:
+                                self.vocab.strings.add(annot[i])
+
+                    if annot is token_annotations[HEAD]["values"]:
+                        for i in range(len(words)):
+                            if attrs.ndim == 1:
+                                attrs[i] = annot[i]
+                            else:
+                                attrs[i, j] = annot[i]
+                    elif annot is token_annotations[MORPH]["values"]:
+                        for i in range(len(words)):
+                            morph_key = self.vocab.morphology.add(token_annotations[MORPH]["values"][i])
+                            if attrs.ndim == 1:
+                                attrs[i] = morph_key
+                            else:
+                                attrs[i, j] = morph_key
+                    else:
+                        for i in range(len(words)):
+                            if attrs.ndim == 1:
+                                attrs[i] = self.vocab.strings[annot[i]]
+                            else:
+                                attrs[i, j] = self.vocab.strings[annot[i]]
+                    j += 1
+            self.from_array(headings, attrs)
+
+        ### Span/document properties ###
+
+        # Complement other document-level properties (cats, spans, ents).
+        self.cats = doc_json.get("cats", self.cats)
+
+        # Set sentence boundaries, if dependency parser not available but sentences are specified in JSON.
+        if not self.has_annotation("DEP"):
+            for sent in doc_json.get("sents", {}):
+                for t in self.char_span(sent["start"], sent["end"]):
+                    self[t.i].is_sent_start = t.i == 0
+
+        if "spans" in doc_json:
+            for span_group in doc_json.get("spans", {}):
+                spans = []
+                for span in doc_json["spans"][span_group]:
+                    char_span = self.char_span(span["start"], span["end"], span["label"], span["kb_id"])
+                    if char_span is None:
+                        raise ValueError(Errors.E1038)
+                    spans.append(char_span)
+                self.spans[span_group] = spans
+
+        if "ents" in doc_json:
+            self.ents = [self.char_span(ent["start"], ent["end"], ent["label"]) for ent in doc_json["ents"]]
```

Comment:
The same `char_span -> None` error can happen here. (Same error message, just with a parameter for `spans`/`ents`.)